acl-lib/src/cmr/acl/core.clj

```diff
@@ -159,21 +159,31 @@
            error-message))
         nil))))
 
+(defn- valid-token?
+  "test"
+  [context]
+  (if-let [token (:token context)]
+    (try
+      (= 200 (first (echo-tokens/get-token-info context token)))
+      (catch Exception e false))
+    true))
+
 (defn has-ingest-management-permission?
   "Returns true if the user identified by the token in the cache has been granted
   INGEST_MANAGEMENT_PERMISSION in ECHO ACLS for the given permission type."
   [context permission-type object-identity-type provider-id]
-  ;; Performance optimization here of returning true if it's the system user.
-  (or (transmit-config/echo-system-token? context)
-      (let [acl-oit-key (echo-acls/acl-type->acl-key object-identity-type)]
-        (->> (get-permitting-acls context
-                                  object-identity-type
-                                  "INGEST_MANAGEMENT_ACL"
-                                  permission-type)
-             ;; Find acls for this provider
-             (filter #(or (nil? provider-id)
-                          (= provider-id (get-in % [acl-oit-key :provider-id]))))
-             seq))))
+  (when (valid-token? context)
```

Comment:
We're losing the performance optimization for the system token. I think it makes sense to always allow the system token to bypass this check.

Edit: I see you are bypassing it in the transmit library.

Reply:
Perhaps a note to that effect here would not go amiss ...